web/client-api/src/main/java/io/deephaven/web/client/api/subscription/SubscriptionType.java

```diff
@@ -0,0 +1,13 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.web.client.api.subscription;
+
+public enum SubscriptionType {
+    /** a subscription that will receive all updates to the table */
+    FULL_SUBSCRIPTION,
+    /** a subscription that will receive updates only for the current viewport */
+    VIEWPORT_SUBSCRIPTION,
```

Comment:
How do growing viewports fit into this model? Not implemented in JSAPI yet?

Reply:
Growing viewports are a server side decision and fit in the category of full subscriptions.